packages/issuer-token-endpoint/lib/IssuerTokenEndpoint.ts

```diff
@@ -0,0 +1,134 @@
+import {
+  AccessTokenResponse,
+  Alg,
+  CNonceState,
+  CredentialOfferState,
+  getNumberOrUndefined,
+  IStateManager,
+  Jwt,
+  JWTSignerCallback,
+  Typ,
+} from '@sphereon/openid4vci-common'
+import express, { NextFunction, Request, Response, Router } from 'express'
+import { v4 } from 'uuid'
+
+const router = express.Router()
+
+interface ITokenEndpointOpts {
+  tokenPath?: string
+  interval?: number
+  cNonceExpiresIn?: number
+  tokenExpiresIn?: number
+  stateManager?: IStateManager<CredentialOfferState>
+  nonceStateManager?: IStateManager<CNonceState>
+  jwtSignerCallback?: JWTSignerCallback
+}
+
+let tokenPath: string
+let interval: number
+let cNonceExpiresIn: number
+let tokenExpiresIn: number
+let stateManager: IStateManager<CredentialOfferState>
+let nonceStateManager: IStateManager<CNonceState>
+let jwtSignerCallback: JWTSignerCallback
+
+export const tokenRequestEndpoint = (opts?: ITokenEndpointOpts): Router => {
+  tokenPath = (opts?.tokenPath ? opts.tokenPath : (process.env.TOKEN_PATH as string)) ?? '/token'
+  interval = (opts?.interval ? opts.interval : getNumberOrUndefined(process.env.INTERVAL)) ?? 300000
+  cNonceExpiresIn = (opts?.cNonceExpiresIn ? opts.cNonceExpiresIn : getNumberOrUndefined(process.env.C_NONCE_EXPIRES_IN)) ?? 300000
+  tokenExpiresIn = (opts?.tokenExpiresIn ? opts.tokenExpiresIn : getNumberOrUndefined(process.env.TOKEN_EXPIRES_IN)) ?? 300000
+  if (opts?.jwtSignerCallback) {
+    jwtSignerCallback = opts.jwtSignerCallback
+  } else {
+    throw new Error('Please provide a private a JWT signer function')
+  }
+  if (opts?.stateManager) {
+    stateManager = opts.stateManager
+  } else {
+    throw new Error('Unable to proceed without an StateManager instance')
+  }
+  if (opts.nonceStateManager) {
+    nonceStateManager = opts.nonceStateManager
+  } else {
+    throw new Error('Unable to proceed without an NonceStateManager instance')
+  }
+  router.post(tokenPath, handleHTTPStatus400, handleTokenRequest)
+  return router
+}
+
+const generateAccessToken = async (): Promise<string> => {
+  const issuanceTime = new Date()
+  const jwt: Jwt = {
+    header: { typ: Typ.JWT, alg: Alg.ES256 },
+    payload: { iat: issuanceTime.getTime(), exp: +new Date(issuanceTime.getTime() + tokenExpiresIn) },
+  }
+  return await jwtSignerCallback(jwt)
+}
+
+const handleTokenRequest = async (request: Request, response: Response) => {
+  response.set({
+    'Cache-Control': 'no-store',
+    Pragma: 'no-cache',
+  })
+
+  const cNonce = v4()
+  await nonceStateManager.setState(cNonce, { cNonce, createdOn: +new Date() })
```

Comment:
Please use seperate values for the nonce (OpenID) and state (OAuth2) values. I am fine with one being the has of the other, or just a seperate uuid.

Reply:
Fixed!